astropy/utils/data.py

```diff
@@ -936,32 +940,37 @@ def get_free_space_in_dir(path):
             "not files.")
         # Actually you can on Linux but I want to avoid code that fails
         # on Windows only.
-    return shutil.disk_usage(path).free
+    free_space = shutil.disk_usage(path).free
+    if return_quantity:
```

Comment:
If we go with the above, then this would become:
```
if unit is not None:
    from astropy import units as u
    free_space = u.Quantity(free_space, u.byte).to(unit)
```